packages/lib-sourcify/src/Verification/Verification.ts

```diff
@@ -0,0 +1,500 @@
+            this.creatorTxHash,
+            creatorTx,
+          );
+        this._onchainCreationBytecode = creationBytecode;
+        this.txIndex = txReceipt.index;
+
+        await this.matchWithCreationTx();
+      } catch (e: any) {
+        logWarn('Error matching with creation tx', {
+          chain: this.sourcifyChain.chainId,
+          address: this.address,
+          creatorTxHash: this.creatorTxHash,
+          error: e.message,
+        });
+      }
+    }
+
+    if (this.creationMatch !== null || this.runtimeMatch !== null) {
+      logInfo('Verified contract', {
+        address: this.address,
+        chainId: this.sourcifyChain.chainId,
+        runtimeMatch: this.runtimeMatch,
+        creationMatch: this.creationMatch,
+      });
+      return;
+    }
+
+    throw new VerificationError(
+      "The deployed and recompiled bytecode don't match.",
+      'NO_MATCH',
+    );
+  }
+
+  async handleSolidityBugCases(
+    forceEmscripten: boolean,
+    compiledRuntimeBytecode: string,
+  ): Promise<SolidityBugType> {
+    // Handle Solidity specific verification bug cases
+    if (
+      this.compilation instanceof SolidityCompilation &&
+      this.runtimeMatch === null
+    ) {
+      const settings = this.compilation.jsonInput.settings;
+
+      // Handle when <0.8.21 and with viaIR and with optimizer disabled
+      // See issues:
+      //   https://github.com/ethereum/sourcify/issues/1088
+      if (
+        !forceEmscripten && // Enter this case only if we are not already forcing Emscripten
+        lt(this.compilation.compilerVersion, '0.8.21') &&
+        !settings.optimizer?.enabled &&
+        settings.viaIR
+      ) {
+        logInfo('Force Emscripten compiler', {
+          address: this.address,
+          chainId: this.sourcifyChain.chainId,
+        });
+
+        // Try to verify again with Emscripten
+        return SolidityBugType.ECMASCRIPT_BUG;
+      }
+
+      // Case when extra unused files in compiler input cause different bytecode
+      // See issues:
+      //   https://github.com/ethereum/sourcify/issues/618
+      //   https://github.com/ethereum/solidity/issues/14250
+      //   https://github.com/ethereum/solidity/issues/14494
+      try {
+        const [, deployedAuxdata] = splitAuxdata(
+          this.onchainRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        const [, recompiledAuxdata] = splitAuxdata(
+          compiledRuntimeBytecode,
+          AuxdataStyle.SOLIDITY,
+        );
+        // Metadata hashes match but bytecodes don't match.
+        if (
+          deployedAuxdata === recompiledAuxdata &&
+          settings.optimizer?.enabled
+        ) {
+          throw new VerificationError(
+            "It seems your contract's metadata hashes match but not the bytecodes. You should add all the files input to the compiler during compilation and remove all others. See the issue for more information: https://github.com/ethereum/sourcify/issues/618",
```

Comment:
This doesn't make sense: "...all all the files ... and remove all others"

We might have to change all occurances of this message in the codebase, but also we should always use error codes instead of messages when checking ie. in the tests
```suggestion
            "It seems your contract's metadata hashes match but not the bytecodes. If you are verifying via metadata.json, use the original full standard JSON input file that has all files including those not needed by this contract. See the issue for more information: https://github.com/ethereum/sourcify/issues/618",
```